server/index.py

```diff
@@ -0,0 +1,333 @@
+        raise Exception("Language not supported")
+
+    # use remote sieve models
+    transcriber = sieve.function.get("sieve/speech_transcriber:8eb4cdf")
+    translator = sieve.function.get("sieve/seamless_text2text:f8695f6")
+    tts = sieve.function.get("sieve/xtts-v1:e788931")
+    lipsyncer = sieve.function.get("sieve/video_retalking:de84f40")
+
+    print("transcribing audio")
+    # transcribe audio
+    transcript = list(transcriber.run(source_audio))
+    text = transcript_to_text(transcript)
+    text_language = "english"
+    print("transcription:", text)
+
+    print("translating text")
+    # Translate text
+    translated_text = translator.run(text, text_language, language)
+    print("translated text:", translated_text)
+
+    print("generating tts audio")
+    # Generate new audio from translated text
+    target_audio = tts.run(translated_text, source_audio, stability=0.5, similarity_boost=0.5)
+    print("done generating audio")
+
+    print("starting lipsync")
+    # Combine audio and video with Retalker
+    return lipsyncer.run(source_video, target_audio).path
+
+@app.route('/projects/<project_id>', methods=['GET'])
+async def get_status(project_id):
+    """Route to return current processing status of a project."""
+    status_file_name = f'{project_id}.txt'
+    status_file_path = os.path.join(get_output_dir_path(), status_file_name)
+    try:
+        with open(status_file_path, "r", encoding="utf-8") as f:
+            data = json.load(f)
+            return jsonify(data), 200
+    except Exception as e:
+        msg = "failed to open status file"
+        print(e, file=sys.stderr)
+        return jsonify({'error': msg}), 500
+
+
+@app.route('/projects/<project_id>/download', methods=['GET'])
+async def download_final_output(project_id):
+    """Route to download final processed output file."""
+    output_file_name = f'{project_id}.mp4'
+    return await send_from_directory(get_output_dir_path(), output_file_name, as_attachment=True)
+
+
+@app.route('/recordings', methods=['GET'])
+async def get_daily_recordings():
+    """Route to fetch all Daily recordings from configured domain."""
+    try:
+        recordings = fetch_recordings()
+        return jsonify({'recordings': recordings}), 200
+
+    except Exception as e:
+        msg = "failed to fetch Daily recordings. Is the Daily API key configured?"
+        print(msg, e, file=sys.stderr)
+        return jsonify({'error': msg}), 500
+
+
+def process_error(msg: str, error: Exception) -> tuple[quart.Response, int]:
+    """Prints provided error and returns appropriately-formatted response."""
+    traceback.print_exc()
+    print(msg, error, file=sys.stderr)
+    response = {'error': msg}
+    return jsonify(response), 500
+
+@app.after_serving
+async def shutdown():
+    for task in app.background_tasks:
+        task.cancel()
+
+def extract_audio(source_video: sieve.Video):
+    import subprocess
+    audio_path = 'temp.wav'
```

Comment:
Won't this break by overwriting the same file over and over if someone initiates two processing requests at the same time? Should probably use a unique file name

Reply:
This was more borrowed code from Sieve's docs, but yes, it will get overwritten, however as the app I built only displays one video transformation at a time, I don't think it's much of a problem if it gets overwritten.

Reply:
Ah ok, and it only lets people _start_ one transformation at a time as well?